lightning/src/ln/channelmanager.rs

```diff
@@ -1980,6 +2013,15 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelMana
 		let onion_packet = onion_utils::construct_onion_packet(onion_payloads, onion_keys, prng_seed, payment_hash);
 
 		let _persistence_guard = PersistenceNotifierGuard::notify_on_drop(&self.total_consistency_lock, &self.persistence_notifier);
+		let mut pending_outbounds = self.pending_outbound_payments.lock().unwrap();
+		let payment_entry = pending_outbounds.entry(payment_id);
+		if let hash_map::Entry::Occupied(payment) = &payment_entry {
+			if !payment.get().is_retryable() {
+				return Err(APIError::RouteError {
+					err: "Payment already completed"
+				});
+			}
```

Comment:
It's unclear to me why this shouldn't be `if payment.get().is_complete() {` as indicated by the error message. Do we want to hit this for legacy payments, too?